schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/avro/AvroFieldsGenerator.java

```diff
@@ -67,36 +70,48 @@ private void parseField(Schema.Field field, List<SchemaFieldInfo> schemaFieldInf
         schemaFieldInfos.add(new SchemaFieldInfo(namespace, name, type.name()));
 
         // todo check whether fields should be mapped to the root schema.
-        parseSchema(schema, schemaFieldInfos);
+        parseSchema(schema, schemaFieldInfos, visitedRecords);
     }
 
-    private void parseSchema(Schema schema, List<SchemaFieldInfo> schemaFieldInfos) {
+    private void parseSchema(Schema schema, List<SchemaFieldInfo> schemaFieldInfos, Set<String> visitedRecords) {
         Schema.Type type = schema.getType();
         LOG.debug("Visiting type: [{}]", type);
 
         switch (type) {
             case RECORD:
-                // store fields of a record.
-                List<Schema.Field> fields = schema.getFields();
-                for (Schema.Field recordField : fields) {
-                    parseField(recordField, schemaFieldInfos);
+
+                String completeName = schema.getFullName();
+
+                if (visitedRecords.contains(completeName)) {
+                    // It seems like the goal of this class is to only return a list of all primitive fields.
+                    //   If we encounter a record that was already parsed, then just ignore it...
```

Comment:
Could you rephrase the comments to "Since we are only interested in primitive data types, if we encounter a record that was already parsed it can be ignored"?. Beginning the comment with "it seems like" sounds speculative.

Reply:
Good point, will change 🙂